code/game/objects/items/syndielad.dm

```diff
@@ -0,0 +1,300 @@
+
+
+/mob/living/carbon/human/key_down(_key, client/user) // STOLEN NANOSUIT CODE
+	switch(_key)
+		if("C")
+			if(istype(gloves, /obj/item/clothing/gloves/syndielad))
+				var/obj/item/clothing/gloves/syndielad/SL = gloves
+				if(SL.syndilad_on && SL.syndilad_ready)
+					SL.syndilad_menu(src)
+				else
+					SL.toggle_power(user)
+				return
+	..()
+
+/obj/item/clothing/gloves/syndielad/proc/check_menu(mob/living/user)
+	if(!user)
+		return FALSE
+	if(user.incapacitated() || !user.Adjacent(src))
+		return FALSE
+	return TRUE
+
+/obj/item/clothing/gloves/syndielad/proc/syndilad_menu(mob/living/user)
+	if(syndilad_detonating)
+		user.visible_message("<span class = 'warning'>[src] rattles violently! It's set to explode!</span>")
+		return FALSE
+	if(syndilad_pinactive)
+		toggle_pinpointer(user)
+	icon_state = "syndilad-on"
+	update_icon()
+	playsound(src, 'sound/items/screwdriver2.ogg', 50, 1)
+	var/list/choices = list(
+	"uplink" = image(icon = 'icons/obj/radio.dmi', icon_state = "radio"),
+	"pinpointer" = image(icon = 'icons/obj/device.dmi', icon_state = "pinpointer"),
+	"poweroff" = image(icon = 'icons/mob/radial.dmi', icon_state = "cable_invalid"),
+	"detonate" = image(icon = 'icons/mob/actions/syndielad.dmi', icon_state = "detonate"),
+	)
+	var/choice = show_radial_menu(user,user, choices, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE)
+	if(!check_menu(user) || !syndilad_ready)
+		return
+	switch(choice)
+		if("uplink")
+			var/datum/component/uplink/uplinkcomponent = GetComponent(/datum/component/uplink)
+			uplinkcomponent.ui_interact(user)
+			icon_state = "syndilad-uplink"
+			playsound(src, 'sound/items/screwdriver2.ogg', 50, 1)
+			return
+		if("pinpointer")
+			toggle_pinpointer(user)
+			playsound(src, 'sound/items/screwdriver2.ogg', 50, 1)
+			return
+		if("poweroff")
+			toggle_power(user)
+			return
+		if("detonate")
+			toggle_detonation(user)
+			return
+
+
+// Nuclear Operative Variant
+
+
+/obj/item/clothing/gloves/syndielad/nuke
+	syndilad_bombmessage = "I'M NUCLEAR"
+
+/obj/item/clothing/gloves/syndielad/nuke/Initialize()
+	. = ..()
+	var/datum/component/uplink/hidden_uplink = GetComponent(/datum/component/uplink)
+	hidden_uplink.set_gamemode(/datum/game_mode/nuclear)
+
+/obj/item/clothing/gloves/syndielad/nuke/scan_for_target()
+	var/obj/item/disk/nuclear/N = locate() in GLOB.poi_list
+	syndilad_target = N
+
+/obj/item/clothing/gloves/syndielad/nuke/examine(mob/user)
+	. = ..()
+	var/msg = "Its tracking indicator reads \"nuclear disk\"."
+	. += msg
+	for(var/obj/machinery/nuclearbomb/bomb in GLOB.machines)
+		if(bomb.timing)
+			. += "Extreme danger. Arming signal detected. Time remaining: [bomb.get_time_left()]."
+
+/obj/item/clothing/gloves/syndielad/nuke/toggle_pinpointer(mob/living/user)
+	syndilad_pinactive = !syndilad_pinactive
+	if(syndilad_pinactive)
+		START_PROCESSING(SSfastprocess, src)
+	else
+		syndilad_target = null
+		STOP_PROCESSING(SSfastprocess, src)
+	update_icon()
```

Comment:
and add `#undef POWEROFF, #undef UPLINK`, etc, etc... down here

Reply:
Made all the changes

Reply:
Eh, you did undef them though so it's probably not a big deal.